### DISCUSSION TITLE
Boolean features

### DISCUSSION THREAD

**Discussion by AKHCE** [MilesCranmer/PySR#436] [Q&A] [answered]
Hi,

I am wondering if there is a way to include Boolean features (i,e, x) in PySR. For example, assume there are 5 categorical features and 2 continuous features, in total 7 features. Is it acceptable to one-hot encode the categorical features and concatenate them with continuous features, and then pass a combination of one-hot and continuous features to PySR?

Thank you so much

**Comment by MilesCranmer** [comment, marked as answer]
Hi @AKHCE,

There is no one-fits-all solution, it will really depend on your problem and what you want. At the end of the day, all calculations will be done using continuous scalars. But indeed you can map categorical/boolean problems into that space without much issue, so boolean problems and categorical problems are totally doable.

There are two things to consider:
1. Accuracy. If you just want accuracy I would try all possible strategies: leaving it as categorical (1, 2, 3, etc)., one-hot, other mappings or combinations.
2. Interpretability. This is entirely dependent on the problem and what you would find interpretable.

Categorical mappings would definitely work if the equation makes sense. For example, for months, you might choose January=1, February=2, and so on. But you could also create a mapping to "phases" of the year by turning it into two features:

$$\text{month} \rightarrow \left( \sin(\frac{2\ \pi\ \text{month}}{12}),\cos(\frac{2\ \pi\ \text{month}}{12}) \right)$$
where $\text{month} \in 1\ldots 12$. This would cause December and January to be close together in any equation without losing each month having a unique number, and avoid having a large jump from 12 to 1.

Cheers,
Miles
